carla/models/catalog/catalog.py

```diff
@@ -0,0 +1,154 @@
+import numpy as np
+
+from ..api import MLModel
+from .load_model import load_model
+
+
+class MLModelCatalog(MLModel):
+    def __init__(
+        self, data_name, model_type, ext="h5", cache=True, models_home=None, **kws
+    ):
+        self._data_name = data_name
+        self._model = load_model(model_type, data_name, ext, cache, models_home, **kws)
+
+        if ext == "pt":
+            self._backend = "pytorch"
+        elif ext == "h5":
+            self._backend = "tensorflow"
+        else:
+            raise Exception("Model type not in catalog")
+
+        self._name = model_type + "_" + data_name
+
+        if data_name == "adult":
```

Comment:
Would be nice if we could find a way to decouple the dataset specifics from the ml model implemenation itself. Could we get this information from the datacatalog instead? (that would probably require to pass the specific datactalog to the MLModel, which could be fine, since model instances are specific the data anyhow?)

Reply:
Still not super happy with hardcoding of datasets/feature inputs here. Could we try to put the feature order somewhere else and only reference it here? (Otherwise we always need to touch this piece of code for every dataset we add) 
- Either deriving it e.g. from the data catalog (e.g. get the column list from the `data.encoded_normalized` - In the future i would envision, thats also the columns that the ML models are trained on? I only temporarily put the previously trained models into `indyfree/cf-models`)
- Or hard-coding it with the models for now in `indyfree/cf-models` and load that data (if we really need it now)

Reply:
If we don't want to store that information together with the underlying mlmodel nor with the data, a simpler solution for now could be to pass the feature input order to the constructor instead?

We can then in a later step retrieve to feature order from configutation or automatically build the feature order. I really think hard-coding it here now is a bad design of the api and diminishes the de-coupling efforts

Reply:
I like the idea about using it as parameter, so every user can define its own feature order. I'll update this in the next commit